server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -707,26 +719,31 @@ private Map<String, Object> createTemplateModel() {
     }
 
     /**
-     * one time operation that extracts the metadata fields from the ingest document and returns them.
-     * Metadata fields that used to be accessible as ordinary top level fields will be removed as part of this call.
+     * Get source and metadata map
      */
-    public Map<Metadata, Object> extractMetadata() {
-        Map<Metadata, Object> metadataMap = new EnumMap<>(Metadata.class);
-        for (Metadata metadata : Metadata.values()) {
-            metadataMap.put(metadata, sourceAndMetadata.remove(metadata.getFieldName()));
-        }
-        return metadataMap;
+    public Map<String, Object> getSourceAndMetadata() {
```

Comment:
We lose the fact that it's an `IngestSourceAndMetadata` if we make the signature a `Map<String, Object>` (which loses access to the typed getters) and if we return an `IngestSourceAndMetadata` then many of the tests for the ingest processors fail to compile because `IngestSourceAndMetadata` is package private, eg

````
modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpProcessorFactoryTests.java:497:
 error: IngestSourceAndMetadata.get(Object) is defined in an inaccessible class or interface                                          
            Map<?, ?> geoData = (Map<?, ?>) ingestDocument.getIngestSourceAndMetadata().get("geoip");  
```

Similar in `org.elasticsearch.action.ingest.AsyncIngestProcessorIT`.